### PR TITLE
Replace `format!` without parameters with `.to_string()`

### PR DESCRIPTION
**Pull request by sbrugman**
## Summary

From https://github.com/astral-sh/ruff/pull/14008/commits/56047ffa53ffd0fb96737b977fcd0612c5103721 I realised that `to_string` is preferred over `format!` when no arguments are provided. This PR consistently does so for existing violations.

## Test Plan

`cargo test`

### CODE REVIEW

File: crates/ruff_linter/src/rules/flake8_pie/rules/unnecessary_placeholder.rs [hunk 1]
```diff
@@ -71,8 +71,8 @@ impl AlwaysFixableViolation for UnnecessaryPlaceholder {
     fn fix_title(&self) -> String {
         let Self { kind } = self;
         match kind {
-            Placeholder::Pass => format!("Remove unnecessary `pass`"),
-            Placeholder::Ellipsis => format!("Remove unnecessary `...`"),
+            Placeholder::Pass => "Remove unnecessary `pass`".to_string(),
+            Placeholder::Ellipsis => "Remove unnecessary `...`".to_string(),
         }
     }
```
Comment (AlexWaygood) [inline crates/ruff_linter/src/rules/flake8_pie/rules/unnecessary_placeholder.rs:77]: here you might be able to do something like

```rs
    fn fix_title(&self) -> String {
        let title = match &self.kind {
            Placeholder::Pass => "Remove unnecessary `pass`",
            Placeholder::Ellipsis => "Remove unnecessary `...`",
        };
        title.to_string()
    }
```

File: crates/ruff_linter/src/rules/flake8_simplify/rules/ast_ifexp.rs [hunk 2]
```diff
@@ -50,9 +50,9 @@ impl Violation for IfExprWithTrueFalse {
     fn fix_title(&self) -> Option<String> {
         let IfExprWithTrueFalse { is_compare } = self;
         if *is_compare {
-            Some(format!("Remove unnecessary `True if ... else False`"))
+            Some("Remove unnecessary `True if ... else False`".to_string())
         } else {
-            Some(format!("Replace with `bool(...)"))
+            Some("Replace with `bool(...)".to_string())
         }
     }
```
Comment (AlexWaygood) [inline crates/ruff_linter/src/rules/flake8_simplify/rules/ast_ifexp.rs:57]: similarly here, we could probably do something like

```rs
    fn fix_title(&self) -> Option<String> {
        let title = if self.is_compare {
            "Remove unnecessary `True if ... else False`"
        } else {
            "Replace with `bool(...)"
        };
        Some(title.to_string())
    }
```

which I would find marginally more readable

**Review by AlexWaygood — APPROVED** [review]
Nice cleanup! A couple of optional nits :-)

File: crates/ruff_linter/src/rules/flake8_pie/rules/unnecessary_placeholder.rs (hunk 1 above)
Comment (sbrugman) [reply crates/ruff_linter/src/rules/flake8_pie/rules/unnecessary_placeholder.rs]: Thanks, changed this according to your suggesions.

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Pull request merged**